generators/app/templates/docs/coding-guides/build-specific-configurations.md

```diff
@@ -0,0 +1,164 @@
+allows mapping each build configuration to a separate environment configuration file for your codebase as well.  It
+also allows for making separate dev-server configurations in case you need to run it differently.
+
+Therefore, each build configuration in the workspace configuration file is a tuple of
+(how-to-build, environment-file-for-codebase), and you'll need a separate configuration for each combination.
+
+## Angular's out-of-the-box environment configuration
+
+### When it works well
+
+This setup works quite well for configuration that's shared among all instances of an environment, like the following
+examples:
+
+- test environment always builds without source maps, disables animations, uses a recaptcha test key, and disables
+  analytics
+- dev environment always builds with source maps, enables animations, uses a recaptcha live key, and disables
+  analytics
+- qa environment always builds with source maps, enables animations, uses a recaptcha live key, and disables
+  analytics
+- prod environment always builds without source maps, enables animations, uses a recaptcha live key, and disables
+  analytics
+
+### Limitations of Angular's fileReplacements
+
+But for certain deployment-specific configuration, things start to get really hairy, like in these examples:
+
+- QA build configuration needs to be built for local deployment, deployment to a server on the internet for QA
+  purposes, and also deployment to another server on the internet for staging purposes
+- Production build needs multiple different deployments of the same app to different servers
+
+These cases can cause problems when:
+
+- Each deployment needs a separate API URL
+- Each deployment needs a separate URL for building its own URLs to where it's deployed
+- Each deployment needs separate API keys, bucket names, etc
+
+You COULD start creating separate configurations for each deployment, each with its own fileReplacements, but that
+would be really messy.
+
+### Workarounds that don't work well
+
+One workaround would be to keep such configurations as globals in a separate deployment-specific script file. But
+that's pretty messy too. More importantly, there are limitations to where they can be used. For example, because
+of AOT, such configuration variables cannot be used in Angular's decorators, because they're not statically
+analyzable. So it would be better if we can keep everything in the same place.
+
+### ngx-rocket to the rescue
+
+The env task solves this problem really well, and avoids the need for separate environment.ts files for
+deployment-specific configuration. 
+
+To add a deployment-specific configuration:
+
+1. edit the existing environment.ts files for whichever environments you'd like to make that variable
+   deployment-specific for by having it come from the imported "env" object - pro tip: you can even make it fall
+   back to an environment-based default and still be statically analyzable!
```

Comment:
^ tbh I've been finding this "pro tip" a little messy - should we nix it?

Reply:
I think it can be a valid use case (though I also don't like it :D ), so it could stays like this in my opinion

Reply:
So I kept it but pushed it off to an "alternatives" section - it made a bit more sense that way after I followed your suggestion to make the tl;dr more to the point :)